src/tools/rbd_mirror/leader_watcher/Types.h

```diff
@@ -32,6 +33,17 @@ struct HeartbeatPayload {
   void dump(Formatter *f) const;
 };
 
+struct HeartbeatAckPayload {
```

Comment:
I was thinking that you could directly use the ack payload sent in response to the ```HeartbeatPayload``` message. That ensures the acks are sent to the leader that actually initiated the message -- and eliminates any potential race during a leader transition.

https://github.com/ceph/ceph/blob/master/src/include/rados/librados.hpp#L1078